addons/l10n_it_edi/tests/test_ir_mail_server.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from unittest.mock import patch
+from collections import namedtuple
+
+from odoo.modules.module import get_module_resource
+from odoo.tests import common
+from odoo import tools
+
+_logger = logging.getLogger(__name__)
+
+
+class MailServerTests(common.TransactionCase):
+    """ Main test class for the l10n_it_edi ir_mail_server class """
+
+    test_content = """<?xml version="1.0" encoding="UTF-8"?>
+<p:FatturaElettronica versione="FPR12" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" 
+xmlns:p="http://ivaservizi.agenziaentrate.gov.it/docs/xsd/fatture/v1.2" 
+xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+xsi:schemaLocation="http://ivaservizi.agenziaentrate.gov.it/docs/xsd/fatture/v1.2 http://www.fatturapa.gov.it/export/fatturazione/sdi/fatturapa/v1.2/Schema_del_file_xml_FatturaPA_versione_1.2.xsd">
+  <FatturaElettronicaHeader></FatturaElettronicaHeader>
+  <FatturaElettronicaBody>
+    <DatiGenerali>
+      <DatiGeneraliDocumento>
+        <TipoDocumento>TD02</TipoDocumento>
+      </DatiGeneraliDocumento>
+    </DatiGenerali>
+  </FatturaElettronicaBody>
+</p:FatturaElettronica>"""
+
+    def setUp(self):
+        """ Setup the test class with a PEC mail server and a fake fatturaPA content """
+        super().setUp()
```

Comment:
Better to use `setUpClass` instead to avoid doing multiple times the same work.

Reply:
Done